File: 2307-replace-non-coprime-numbers-in-array/replace-non-coprime-numbers-in-array.java
class Solution {
    public static int gcd(int a,int b){
        if(a==0)return b;
        if(b==0)return a;
        if(a>b)return gcd(a%b,b);
         return gcd(a,b%a);
         

    }
    public static int lcm(int a,int b){
        int t=gcd(a,b);
         long pro=1L*a*b;
         return (int)(pro/t);
    }
    public List<Integer> replaceNonCoprimes(int[] nums) {
        Stack<Integer>st=new Stack<>();
        for(int i=0;i<nums.length;i++){
            int curr=nums[i];
            while(!st.empty()){
                int t=st.peek();
                int GCD=gcd(t,curr);
                if(GCD==1)break;
                st.pop();
                curr=lcm(t,curr);
            }
            st.push(curr);
        }
       List<Integer> arr=new ArrayList<>();
        while(!st.empty()){
            arr.add(st.peek());
            st.pop();
        }
        Collections.reverse(arr);
        return arr;
    
}
}

File: 2307-replace-non-coprime-numbers-in-array/replace-non-coprime-numbers-in-array.cpp
class Solution {
public:
    vector<int> replaceNonCoprimes(vector<int>& nums) {
        vector<int>ans;
       
        for(int i=0;i<nums.size();i++){
            int curr=nums[i];
            while(!ans.empty()){
            int t=ans.back();
            int GCD=gcd(t,curr);
            if(GCD==1)break;
                ans.pop_back();
                curr=lcm(t,curr);
               
            }
           
            
            ans.push_back((int)curr);
        }
        
      return ans;
        
    }
};